### DISCUSSION TITLE
3D krigging

### DISCUSSION THREAD

**Discussion by ELEGER** [GeoStat-Framework/GSTools#301] [Q&A]
Hello,
First of all I would like to thank the GS team for the very rich documentation available. I especially really enjoyed the documentation and examples available on readthedocs and have a big crush for the WellTestPy and AnaFlow packages.

As the title gives a hint, I would like precision on which implementation is faster/more robust for kriging in 3d (lat/long/elevation + Values). I got pretty lost on the GS or Pykrige preferences, especially after finding some [opinions](https://gis.stackexchange.com/questions/379516/rule-of-thumb-for-complexity-of-3d-kriging-with-pykrige). Are pykrige and Gs -tools based on the sames routines ?

On a side note, I would like to perform 3D krigging of Geophysical data (electrical resistivity), and I'm wondering how I could give anysotropical/preferential direction during the krigging. For instance a fault.

With all the best
Emmanuel

**Comment by ELEGER** [comment]
Hello,
Any idea ?
Cheers ?
e.

**Comment by LSchueler** [comment]
Hi Emmanuel,

I'm really sorry for the very late reply.
Did you already perform the krigging? Does the computation take too long? - I would suggest first trying out the GSTools routines with parallelisation or even the Rust-core activated, have a look at the [installation guide](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/index.html) subsections **Speeding up GSTools by parallelization** and **Using experimental GSTools-Core for even more speed**.

And regarding your second question, you can simply provide an anisotropic covariance model to the kriging routine. [Here](https://geostat-framework.readthedocs.io/projects/gstools/en/stable/examples/01_random_field/03_unstr_srf_export.html#sphx-glr-examples-01-random-field-03-unstr-srf-export-py) is a very simple example, which uses an anisotropic covariance model to generate a spatial random field.
